expression/builtin_cast_vec.go

```diff
@@ -1005,11 +1005,37 @@ func (b *builtinCastJSONAsDurationSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinCastDecimalAsJSONSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDecimalAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveJSON(n)
+	f64s := buf.Decimals()
+	var f float64
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendJSON(json.BinaryJSON{})
+			result.SetNull(i, true)
```

Comment:
The JSON type is not fixed-length, although these code works fine for now, 
I think we should call `result.AppendNull` here.

Reply:
OK, I had changed it.